src/seedu/addressbook/model/person/UniquePersonList.java

```diff
@@ -0,0 +1,129 @@
+package seedu.addressbook.model.person;
+
+import seedu.addressbook.Utils;
+import seedu.addressbook.model.DuplicateDataException;
+
+import java.util.*;
+
+/**
+ * A list of persons that enforces no nulls and uniqueness between its elements.
+ *
+ * Supports minimal set of list operations needed for the app's features.
+ *
+ * @see Person#equals(Object)
+ * @see Utils#elementsAreUnique(Collection)
+ */
+public class UniquePersonList implements Iterable<Person> {
```

Comment:
I agree; however when we move to javaFX, the view uses `ObservableList` and having to propogate changes in the set to the obs-list negates most of the time complexity advantages of using a set. 

So should we keep it as a list? (easier migration to javafx based level 3)
or use a set? (semantically better, but has to propogate changes to list view)